lib/cadet_web/controllers/assessments_controller.ex

```diff
@@ -0,0 +1,164 @@
+    response(401, "Unauthorised")
+  end
+
+  swagger_path :show do
+    get("/assessments/{assessmentId}")
+
+    summary("Get information about one particular assessment.")
+
+    security([%{JWT: []}])
+
+    produces("application/json")
+
+    parameters do
+      assessmentId(:path, :integer, "assessment id", required: true)
+    end
+
+    response(200, "OK", Schema.ref(:Assessment))
+    response(400, "Missing parameter(s) or invalid assessmentId")
+    response(401, "Unauthorised")
+  end
+
+  def swagger_definitions do
+    %{
+      AssessmentsList:
+        swagger_schema do
+          description("A list of all assessments")
+          type(:array)
+          items(Schema.ref(:AssessmentOverview))
+        end,
+      AssessmentOverview:
+        swagger_schema do
+          properties do
+            id(:integer, "The assessment id", required: true)
+            title(:string, "The title of the assessment", required: true)
+            type(:string, "Either mission/sidequest/path/contest", required: true)
+            summary_short(:string, "Short summary", required: true)
+            open_at(:string, "The opening date", format: "date-time", required: true)
+            close_at(:string, "The closing date", format: "date-time", required: true)
+
+            max_xp(
+              :integer,
+              "The maximum amount of XP to be earned from this assessment",
+              required: true
+            )
+
+            cover_picture(:string, "The URL to the cover picture", required: true)
+          end
+        end,
+      Assessment:
+        swagger_schema do
+          properties do
+            id(:integer, "The assessment id", required: true)
+            title(:string, "The title of the assessment", required: true)
+            type(:string, "Either mission/sidequest/path/contest", required: true)
+            summary_long(:string, "Long summary", required: true)
+            mission_pdf(:string, "The URL to the assessment pdf")
+
+            questions(Schema.ref(:Questions), "The list of questions for this assessment")
+          end
+        end,
+      Questions:
+        swagger_schema do
+          description("A list of questions")
+          type(:array)
+          items(Schema.ref(:Question))
+        end,
+      Question:
+        swagger_schema do
+          properties do
+            questionId(:integer, "The question id", required: true)
+            questionType(:string, "The question type (mcq/programming)", required: true)
+            content(:string, "The question content", required: true)
+
+            choices(
+              Schema.new do
+                type(:array)
+                items(Schema.ref(:MCQChoice))
+              end,
+              "MCQ choices if question type is mcq"
+            )
+
+            solution(:integer, "Solution to a mcq question if it belongs to path assessment")
+
+            answer(
+              :string_or_integer,
```

Comment:
@indocomsoft I was looking at the Grading specs and saw this. An issue with swagger perhaps? 
![image](https://user-images.githubusercontent.com/12512488/42256555-a8e49e40-7f84-11e8-874a-9cf8215e83bc.png)

Reply:
Also, I think questionType and questionId should be changed to snake_case. I use camelCase for Typescript so that's probably where it got mixed in